scripts/generate-docs.js

```diff
@@ -266,27 +292,45 @@ if (require.main === module) {
 	require("./_command")({
 		exec: generate,
 		async watch() {
-			let timeout, genPromise
-			function updateGenerator() {
-				if (timeout == null) return
-				clearTimeout(timeout)
-				genPromise = new Promise((resolve) => {
-					timeout = setTimeout(function() {
-						timeout = null
-						resolve(makeGenerator().then((g) => g.generate()))
-					}, 100)
-				})
+			const generator = (await makeGenerator())
+
+			function isLayoutFile(relative) {
+				return relative === "layout.html"
+			}
+
+			function isNavigationFile(relative) {
+				return ["nav-guides.md", "nav-methods.md"].includes(relative)
+			}
+
+			async function updateGenerator() {
+				generator.generate()
 			}
 
 			async function updateFile(file) {
-				if ((/^layout\.html$|^archive$|^nav-/).test(file)) {
-					updateGenerator()
+				const relative = path.relative(r("docs"), file)
+				if (isLayoutFile(relative)) {
+					generator.setLayout(await loadLayoutFile())
+					generator.generate()
+				} else if (isNavigationFile(relative)) {
+					const [guides, methods] = await loadNavigationFiles()
+					generator.setGuides(guides)
+					generator.setMethods(methods)
+					generator.generate()
+				} else {
+					generator.generateSingle(file)
```

Comment:
Every single one of those `.generate()` calls need `await`s as they all return promises.

```suggestion
					generator.setLayout(await loadLayoutFile())
					await generator.generate()
				} else if (isNavigationFile(relative)) {
					const [guides, methods] = await loadNavigationFiles()
					generator.setGuides(guides)
					generator.setMethods(methods)
					await generator.generate()
				} else {
					await generator.generateSingle(file)
```

Reply:
fixed